### DISCUSSION TITLE
Breaking Change in 5.23.0 with rename of DriveItem.FileObject and semver?

### DISCUSSION THREAD

**Discussion by rominator1983** [microsoftgraph/msgraph-sdk-dotnet#2129] [General]
Wanted to ask if versioning should strictly adhere to the semver guidelines for this repository.
Recently our build broke with the automated upgrade to 5.23.0 as DriveItem.FileObject was renamed to File. This should not happen considering semver rules.

**Comment by andrueastman** [comment]
Thanks for raising this @rominator1983,

Your expectation of functionality in upcoming versions would align with what we do in the future. The naming of the properties as `FileObject` rather than `File` was [considered a bug with the code](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/CHANGELOG.md#5230---2023-08-16) and not really change which is why the version was not moved up in this case.

**Reply by rominator1983** [reply]
I see. Would have been a good point for the release notes though (or aging out via obsolete attribute) since the change violates semver nonetheless. 

Thanks anyhow of course for the great work and your quick reply. I basically wanted to know, if we are on the same page here which we are.

Thank you
